src/pint/residuals.py

```diff
@@ -591,6 +611,15 @@ def chi2(self):
             chi2 += res.chi2
         return chi2
 
+    @property
+    def dof(self):
+        dof = len(self._combined_resids)
+        # It assumes that the input model are the same model, and time residual has
+        # offset in the fitting
+        # TODO In a more general case, this assumption would not be valid.
+        dof -= len(self.residual_objs["toa"].model.free_params) + 1
```

Comment:
Shouldn't `self` have its own `.model` attribute?

Reply:
Is this explanation clear?

Reply:
I would view that as one model with two parts. But more to the point, here and now, as a user, I would expect every Residuals object, Combined or not, to have a `.model` attribute.

Reply:
I can make an property that takes the models from all the residuals. But it would not be one model object.  I think the assumption that all residual objects have the same model may cause problem in the future.  On the other hand, if we want to view the it as one model with two parts. I think a better way to do this is to make another residual type, that defined to have one model but used in the multiple residuals small classes.

Reply:
I think the individual contained residuals shouldn't have `model` attributes either, but that's less important - if we want to do a joint fit, speculatively in the future, we can build a joint model, whose free parameters are assembled from the free parameters of the sub-models, with whatever linkages. But for now I think the invariant that "every Residual object you get from a model keeps track of that model in its `.model` attribute" is a good one to maintain.